src/scripts/components/VariableMessage.jsx

```diff
@@ -18,24 +20,43 @@ import {
 } from '../constants';
 
 class VariableMessage extends React.Component {
+  currentDraggedSpanVariable = '';
+  isCategoryMobileViewVisible = false;
   state = {
     message: '',
     available: [],
+    variables: [],
+    variablesOfCategory: [],
+    categories: [],
+    selectedCategory: '',
   };
 
   componentWillMount() {
+    const toUpdateState = {
+      categories: [],
+    };
+
+    if (this.props.isCategoryAvailable) {
+      this.isCategoryMobileViewVisible = window.outerWidth <= 767; //  for mobile view
+      toUpdateState.selectedCategory = this.props.defaultSelectedCategory;
+      toUpdateState.variablesOfCategory =
+        this.props.data.filter((item) => {
```

Comment:
Seems here we're using a `filter` function as an iterating loop to get properties into an array that is unrelated to the item at hand. Would it not be easer to just filter based on category here and handle adding up the categories in a separate line for clarity?

E.G:
```
const variablesOfSelectedCategory = this.props.data.filter(item => item.category === this.props.selectedCategory);

const categories = [...new Set(this.props.data.map(item => item.category))]; // Set to remove duplicates.
```

Reply:
Yeah this looks cool, but acc. to optimization here we are running loop `this.props.data` 2 times. As in above we are doing in single loop in above. Let me know what u think.

Reply:
Yeah Sure. Done.